### PR TITLE
Implement UserListToolbar component

### PR DESCRIPTION
**Pull request by EVAST9919**
Screenshot_2

### CODE REVIEW

File: osu.Game.Tests/Visual/UserInterface/TestSceneUserListToolbar.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+            });
+
+            Add(new FillFlowContainer
+            {
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(0, 5),
```
Comment (bdach) [inline osu.Game.Tests/Visual/UserInterface/TestSceneUserListToolbar.cs]: Not sure if this shouldn't be
```suggestion
                Spacing = new Vector2(0, 10),
```
Each sort tab item on web has 5px margin individually, which comes to 10px spacing between items altogether.

File: osu.Game/Overlays/Comments/CommentsHeader.cs [hunk 2]
```diff
@@ -11,6 +11,8 @@
 using osuTK;
 using osu.Framework.Input.Events;
 using osu.Game.Graphics.Sprites;
+using System.ComponentModel;
+using Container = osu.Framework.Graphics.Containers.Container;
```
Comment (bdach) [inline osu.Game/Overlays/Comments/CommentsHeader.cs]: It's a nitpick but I'd prefer this the other way round:
```csharp
using osu.Framework.Graphics.Containers;
using Description = System.ComponentModel.Description;
```
or even use the fully qualified name.

File: osu.Game/Overlays/OverlayPanelDisplayStyleControl.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            });
+        }
+
+        protected override TabFillFlowContainer CreateTabFlow() => new TabFillFlowContainer
+        {
+            AutoSizeAxes = Axes.Both,
+            Direction = FillDirection.Horizontal
+        };
+
+        private class PanelDisplayTabItem : TabItem<OverlayPanelDisplayStyle>, IHasTooltip
```
Comment (bdach) [inline osu.Game/Overlays/OverlayPanelDisplayStyleControl.cs]: These should probably have `HoverClickSounds` for UX consistency, to indicate interactability.

**Review by bdach — COMMENTED** [review]
The *Rank* button does not have a ruleset icon like web does:

image

This is generally probably going to be a bigger problem if we're aiming for 100% parity with web as the ruleset is taken from the user's selected primary ruleset on their profile. I'd be fine with it being a bindable for now if we decide to include it and populate later when that information is available from API (if it isn't already).

File: osu.Game/Overlays/OverlayPanelDisplayStyleControl.cs (hunk 3 above)
Comment (EVAST9919) [reply osu.Game/Overlays/OverlayPanelDisplayStyleControl.cs]: oops, true

**Comment by EVAST9919** [comment]
Also the icon require some refactoring in the `OverlaySortTabControl` and pr will be more confusing.

**Comment by bdach** [comment]
OK, if that is the case then we can go back to that after the eventual refactor.

File: osu.Game.Tests/Visual/UserInterface/TestSceneUserListToolbar.cs (hunk 1 above)
Comment (EVAST9919) [reply osu.Game.Tests/Visual/UserInterface/TestSceneUserListToolbar.cs]: You selected wrong file. In `UserListToolbar` it has 10px spacing

**Review by bdach — APPROVED** [review]
Looks good for now.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**